internal/conf/init.go

```diff
@@ -6,9 +6,18 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/httpcli"
 )
 
+// Init function must be called by every service that requires access to components in conf package.
+// If the caller is in the client mode, we start a goroutine continuously polling for config changes.
```

Comment:
```suggestion
// Init function completes the initialization process of the conf package, starting the configuration continuous changes polling if in client mode. The conf.Watch function can safely be called before calling Init to register callbacks reacting to the changes. 
// 
// The Init function must be called early in an application initialization process, but tests do not need to call it. 
```

Reasoning about the changes 😊: 

- The fact it uses a goroutine is an implementation detail, so we don't need to talk about it. 
- It's quite common to start a docstring in Go with what the function does, then to state if there are any contextual things to know, which is why I moved the fact that it needs to be called by all apps at the end. 
- It not be clear for the reader what can be called before this, which is why we explicit the `conf.Watch` part. Same logic with the tests!

Reply:
Love it, thanks!